common/src/VisVehicleHal.cpp

```diff
@@ -0,0 +1,629 @@
+        }
+        ALOGD("Found mapped property 0x%x area 0x%x to vis %s", propValue.prop, propValue.areaId,
+              it->second.c_str());
+        std::stringstream aa;
+        aa << "\"" << it->second << "\":" << vehiclePropValueToString(propValue);
+        epam::Status st = mVisClient.setPropertySync(aa.str());
+        if (st != epam::Status::OK) {
+            ALOGE("SET retrned != OK [%d]", st);
+            return StatusCode::TRY_AGAIN;
+        }
+    }
+    if (!mPropStore->writeValue(propValue, false)) {
+        return StatusCode::INVALID_ARG;
+    }
+
+    return StatusCode::OK;
+}
+
+std::vector<VehiclePropConfig> VisVehicleHal::listProperties() const {
+    return mPropStore->getAllConfigs();
+}
+
+StatusCode VisVehicleHal::subscribe(int32_t property, float sampleRate) {
+    ALOGI("%s propId: 0x%x, sampleRate: %f", __func__, property, sampleRate);
+    fetchAllAndSubscribe();
+    if (isContinuousProperty(property)) {
+        mRecurrentTimer.registerRecurrentEvent(hertzToNanoseconds(sampleRate), property);
+    }
+    return StatusCode::OK;
+}
+
+StatusCode VisVehicleHal::unsubscribe(int32_t property) {
+    ALOGI("%s propId: 0x%x", __func__, property);
+    if (isContinuousProperty(property)) {
+        mRecurrentTimer.unregisterRecurrentEvent(property);
+    }
+    return StatusCode::OK;
+}
+
+void VisVehicleHal::subscriptionHandler(const epam::CommandResult& result) {
+    for (auto& item : result) {
+        /* Several vehicle properties may be mapped to one VIS property. Will find & update all of
+         * these. */
+        auto range = mVisNameToVProperty.equal_range(item.first);
+        for (auto it = range.first; it != range.second; ++it) {
+            ALOGV("Will convert VIS property %s to vehicle prop", item.first.c_str());
+            auto result = mPropStore->readValueOrNull(it->second.prop, it->second.area);
+            if (result) {
+                auto val = result.get();
+                if (jsonToVehicle(item.second, *val)) {
+                    if (mPropStore->writeValue(*val, true)) {
+                        ALOGV("Value for property %d area=%d|%s updated to %s", it->second.prop,
+                              it->second.area, it->first.c_str(),
+                              vehiclePropValueToString(*val).c_str());
+                        auto v = getValuePool()->obtain(*val);
+                        v->timestamp = elapsedRealtimeNano();
+                        doHalEvent(std::move(v));
+                    } else {
+                        ALOGE("Unable to update property %d area=%d|%s", it->second.prop,
+                              it->second.area, it->first.c_str());
+                    }
+                }
+            } else {
+                ALOGE("Unable to read current value for prop 0x%x area=0x%x from propertystore",
+                      it->second.prop, it->second.area);
+            }
+        }
+    }
+#if 0
```

Comment:
Do we need to keep disabled code?
If it is intended to be used in some cases, may be to add some comments regarding possible use case.

Reply:
In short - yes. We need it for debug/simulation.